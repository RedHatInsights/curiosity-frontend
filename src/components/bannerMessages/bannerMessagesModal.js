import React from 'react';
import {
  AlertActionLink,
  Button,
  Modal,
  ModalBody,
  ModalFooter,
  ModalHeader,
  ModalVariant
} from '@patternfly/react-core';
import { translate } from '../i18n/i18nHelpers';

/**
 * @memberof BannerMessages
 * @module BannerMessagesModal
 */

/**
 * A self-contained modal wrapper for banner messages.
 *
 * @param {object} props
 * @param {React.ReactNode} props.modalTitle
 * @param {React.ReactNode} [props.modalContent]
 * @param {React.ReactNode} props.children
 * @param {translate} [props.t=translate]
 * @returns {JSX.Element}
 */
const BannerMessagesModal = ({ modalTitle, modalContent = '...', children, t = translate }) => {
  const [isOpen, setIsOpen] = React.useState(false);

  const onClose = () => {
    setIsOpen(false);
  };

  const onClick = () => {
    setIsOpen(true);
  };

  return (
    <React.Fragment>
      <Modal
        appendTo={document.querySelector('.curiosity')}
        className="curiosity-banner-modal"
        onClose={onClose}
        isOpen={isOpen}
        variant={ModalVariant.small}
      >
        <ModalHeader title={modalTitle} />
        <ModalBody>
          <div data-test="bannerMessageModalContent">{modalContent}</div>
        </ModalBody>
        <ModalFooter>
          <Button key="confirm" variant="secondary" onClick={onClose}>
            {t('curiosity-banner.modal', { context: ['label', 'close'] })}
          </Button>
        </ModalFooter>
      </Modal>
      <AlertActionLink data-test="bannerMessageModalLink" isInline component="a" onClick={onClick}>
        {children}
      </AlertActionLink>
    </React.Fragment>
  );
};
export { BannerMessagesModal as default, BannerMessagesModal };
