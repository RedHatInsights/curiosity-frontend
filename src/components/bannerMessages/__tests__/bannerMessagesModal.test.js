import React from 'react';
import { BannerMessagesModal } from '../bannerMessagesModal';

describe('BannerMessagesModal Component', () => {
  it('should render a basic component', async () => {
    const props = {
      modalTitle: 'Lorem ipsum'
    };

    const component = await shallowComponent(<BannerMessagesModal {...props} />);
    expect(component).toMatchSnapshot('basic');
  });

  it('should handle opening a modal', () => {
    const props = {
      modalTitle: 'Lorem ipsum'
    };

    const component = renderComponent(
      <div>
        <div className="curiosity" />
        <BannerMessagesModal {...props}>Click me</BannerMessagesModal>
      </div>
    );
    component.fireEvent.click(component.find('a'));
    expect(component.screen.render()).toMatchSnapshot('open');
  });
});
