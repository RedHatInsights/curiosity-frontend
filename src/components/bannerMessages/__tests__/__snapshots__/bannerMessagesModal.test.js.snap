// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerMessagesModal Component should handle opening a modal: open 1`] = `
<screen>
  <div
    class="pf-v6-c-backdrop"
    id="pf-modal-part-1"
  >
    <div
      class="pf-v6-l-bullseye"
    >
      <div
        aria-labelledby="pf-modal-part-0"
        aria-modal="true"
        class="pf-v6-c-modal-box curiosity-banner-modal pf-m-sm"
        data-ouia-component-id="OUIA-Generated-Modal-small-1"
        data-ouia-component-type="PF6/ModalContent"
        data-ouia-safe="true"
        id="pf-modal-part-0"
        role="dialog"
      >
        <div
          class="pf-v6-c-modal-box__close"
        >
          <button
            aria-label="Close"
            class="pf-v6-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Modal-small-1-ModalBoxCloseButton"
            data-ouia-component-type="PF6/Button"
            data-ouia-safe="true"
            type="button"
          >
            <span
              class="pf-v6-c-button__icon"
            >
              <svg
                aria-hidden="true"
                class="pf-v6-svg"
                fill="currentColor"
                height="1em"
                role="img"
                viewBox="0 0 352 512"
                width="1em"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                />
              </svg>
            </span>
          </button>
        </div>
        <header
          class="pf-v6-c-modal-box__header"
        >
          <h1
            class="pf-v6-c-modal-box__title"
          >
            <span
              class="pf-v6-c-modal-box__title-text"
            >
               Lorem ipsum 
            </span>
          </h1>
        </header>
        <div
          class="pf-v6-c-modal-box__body"
        >
          <div
            data-test="bannerMessageModalContent"
          >
             ... 
          </div>
        </div>
        <footer
          class="pf-v6-c-modal-box__footer"
        >
          <button
            class="pf-v6-c-button pf-m-secondary"
            data-ouia-component-id="OUIA-Generated-Button-secondary-1"
            data-ouia-component-type="PF6/Button"
            data-ouia-safe="true"
            type="button"
          >
            <span
              class="pf-v6-c-button__text"
            >
               t(curiosity-banner.modal_label, {"context":"close"}) 
            </span>
          </button>
        </footer>
      </div>
    </div>
  </div>
</screen>
`;

exports[`BannerMessagesModal Component should render a basic component: basic 1`] = `
<React.Fragment>
  <Modal
    appendTo={null}
    className="curiosity-banner-modal"
    isOpen={false}
    onClose={[Function]}
    ouiaSafe={true}
    position="default"
    variant="small"
  >
    <ModalHeader
      title="Lorem ipsum"
    />
    <ModalBody>
      <div
        data-test="bannerMessageModalContent"
      >
        ...
      </div>
    </ModalBody>
    <ModalFooter>
      <Button
        onClick={[Function]}
        variant="secondary"
      >
        t(curiosity-banner.modal_label, {"context":"close"})
      </Button>
    </ModalFooter>
  </Modal>
  <AlertActionLink
    component="a"
    data-test="bannerMessageModalLink"
    isInline={true}
    onClick={[Function]}
  />
</React.Fragment>
`;
